docs/developer-guide/storage/README.md

```diff
@@ -0,0 +1,240 @@
+---
+order: 6
+---
+# Data storage and processing
+
+Kestra's primary purpose is to orchestrate data processing via tasks, so data is central to each flow execution.
+
+Depending on the task, data can be stored inside the execution context or inside Kestra's internal storage. You can also manually store data inside Kestra's state store by using dedicated tasks.
+
+Some tasks give you the choice of where you want to store the data, usually using a `fetchType` property or the three `fetch`/`fetchOne`/`store` properties.
+
+For example, using the DynamoDB Query task:
+
+```yaml
+id: query
+type: io.kestra.plugin.aws.dynamodb.Query
+tableName: persons
+keyConditionExpression: id = :id
+expressionAttributeValues:
+  :id: "1"
+fetchType: FETCH
+```
+
+The `fetchType` property can have four values:
+- `FETCH_ONE`: will fetch the first row and set it in a task output attribute (the `row` attribute for DynamoDB); the data will be stored inside the execution context.
+- `FETCH`: will fetch all rows and set them in a task output attribute (the `rows` attribute for DynamoDB); the data will be stored inside the execution context.
+- `STORE`: will store all rows inside Kestra's internal storage. The internal storage will return a URI usually set in the task output attribute `uri` and that can be used to retrieve the file from the internal storage.
+- `NONE`: will do nothing.
+
+The three `fetch`/`fetchOne`/`store` properties will do the same but using three different task properties instead of a single one.
+
+## Storing data
+
+### Storing data inside the execution context
+
+Data can be stored as variables inside the execution context. This can be convenient for sharing data between tasks.
+
+To do so, tasks store data as output attributes that are then available inside the flow via Pebble expressions like <code v-pre>{{outputs.taskName.attributeName}}</code>. More information about outputs can be found [here](../outputs/).
+
+Be careful that when the size of the data is significant, this will increase the size of the execution context, which can lead to slow execution and increase the size of the execution storage inside Kestra's repository. 
+
+::: warning
+When using Kafka as Kesta's internal queue, a limit on the size of the execution exists, which is the limit of the message that can be sent to Kafka. In this setup, this is important to avoid storing large amounts of data inside the execution context.
```

Comment:
Well, there is no limit when using the database so I can say something like "dependending on your repository implementation a size limit may exist" ... But here I wanted to directly raise a warning size about Kafka and big execution context.

Reply:
The big execution limit exists also on the database (since the performance will be degraded with all). 
I love the idea to say 1M as an hard limit even if it's not true, it will help people understand the maximum size (and maybe add a precision on that this limit is on execution and not each task)

Reply:
OK, I can say "some internal repositories have an hard limit on the size of a message, others not, but in anycase we strongly advise not to have an execussion more than 1MB.
I'll clarify it's for the execution of a flow

Reply:
WDYT of 

> ::: warning
Depending on the Kestra internal queue and repository implementation, there can be a hard limit on the size of the flow execution context as it is stored as a single row/message. Usually, this limit is around 1MB, so this is important to avoid storing large amounts of data inside the flow execution context.
:::

Reply:
perfect :)

Reply:
updated!